src/Nethermind/Nethermind.Evm/IntrinsicGasCalculator.cs

```diff
@@ -80,4 +81,22 @@ private static long AccessListCost(Transaction transaction, IReleaseSpec release
 
         return accessListCost;
     }
+
+    private static long AuthorizationListCost(Transaction transaction, IReleaseSpec releaseSpec)
+    {
+        AuthorizationTuple[]? authorizationList = transaction.AuthorizationList;
+        long contractCodeCost = 0;
+        if (authorizationList is not null)
+        {
+            if (!releaseSpec.IsAuthorizationListEnabled)
+            {
+                throw new InvalidDataException(
```

Comment:
Should we throw here? This isn't validation?

If not this simplifies to:
```
  return (transaction.AuthorizationList?.Length ?? 0) * GasCostOf.PerAuthBaseCost;
```

Reply:
> Should we throw here? This isn't validation?
> 
> If not this simplifies to:
> 
> ```
>   return (transaction.AuthorizationList?.Length ?? 0) * GasCostOf.PerAuthBaseCost;
> ```

`AccessListCost` will throw if the spec doesn't allow it, thats why i did the same.